be/src/exec/vectorized/hdfs_scanner_text.h

```diff
@@ -11,7 +11,11 @@ namespace starrocks::vectorized {
 class HdfsTextScanner final : public HdfsScanner {
 public:
     HdfsTextScanner() = default;
-    ~HdfsTextScanner() override = default;
+    ~HdfsTextScanner() override {
+        if (_runtime_state != nullptr) {
```

Comment:
Could we invoke `close(_runtime_state)` on `HdfsScanner::~HdfsScanner` so that we don't have to invoke it on every derived class

Reply:
No. I've tried that before but failed. Reason is  in `close`method a virtual method `do_close` is called.  And you will get exception of `pure virtual method called` exception. Think this case, when you call `~HdfsParquetScanner`

- `~HdfsParquetScanner`
- `~HdfsScanner`
- `close`
- `do_close`

But at this time, this `do_close` is not method of `HdfsParquetScanner`, but method of `HdfsScanner`.  So you get a exception of `pure virtual method called`.

Reply:
I really wish that this method can be called automatically at base class, instead of same routine in every sub classes.

Reply:
But in theory this clsoe method will be called by `HdfsScanNode`. And I put close in dtor is simply for safety. Better to remove it, otherwise it might confuse people. What do you think?

Reply:
Agree, remove it is better

Reply:
> In ideal case, this clsoe method will be called by `HdfsScanNode`. But there is a concern that, during execution some exception might happens and dtor can make sure `close` is called to avoid resource leak.

Yeah, maybe HdfsScanNode has ability of RAII and release resource of HdfsScanner properly.

Reply:
After second thought, I think maybe it's better to keep it, because safety comes first.

Reply:
After second thought, I think maybe it's better to keep it, because safety comes first. And I extract a common method `cleanup` to do that.